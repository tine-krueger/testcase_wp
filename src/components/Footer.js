import styled from 'styled-components/macro'

export default function Footer() {
    return(
        <BlackFooter>
            <p>Web Dev Testcase</p>
            <p>Â©2021 Wechselpilot</p>
            <p>No rights reserved</p>
        </BlackFooter>
    )
}

const BlackFooter = styled.div`
    padding: 1em 0;
    width: 100%;
    position: absolute;
    bottom: -1em;
    left: 0;
    display: grid;
    background: #000;
    color: #fff;

    p {
        justify-self: center;
        text-transform: uppercase;
        margin: .5em 0;
    }

`