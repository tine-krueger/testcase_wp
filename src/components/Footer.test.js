import { render } from '@testing-library/react'
import 'jest-styled-components'
import Footer from './Footer'

describe ('Footer', () => {
    it('renders correctly', () => {
        const { container } = render(
            <Footer />
        )
        expect(container).toMatchSnapshot()
    })

    it('shows the correct text', () => {
        const { getByText} = render(
            <Footer />
        )
        expect(getByText(/Web Dev Testcase/i)).toBeInTheDocument()
        expect(getByText(/Â©2021 Wechselpilot/i)).toBeInTheDocument()
        expect(getByText(/No rights reserved/i)).toBeInTheDocument()
    })
})