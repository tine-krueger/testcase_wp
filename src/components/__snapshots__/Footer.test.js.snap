// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
.c0 {
  padding: 1em 0;
  width: 100%;
  position: absolute;
  bottom: -.2em;
  left: 0;
  display: grid;
  background: var(--footerBackground);
  color: var(--secondaryFont);
}

.c0 p {
  justify-self: center;
  text-transform: uppercase;
  margin: .5em 0;
}

@media (min-width:992px) {
  .c0 {
    grid-template-columns: 1fr 1fr 1fr;
    height: 4em;
  }

  .c0 p {
    -webkit-align-self: center;
    -ms-flex-item-align: center;
    align-self: center;
  }

  .c0 p:first-child {
    justify-self: end;
  }

  .c0 p:last-child {
    justify-self: start;
  }
}

<div>
  <div
    class="c0"
  >
    <p>
      Web Dev Testcase
    </p>
    <p>
      Â©2021 Wechselpilot
    </p>
    <p>
      No rights reserved
    </p>
  </div>
</div>
`;
